pyccel/codegen/printing/ccudacode.py

```diff
@@ -344,15 +344,15 @@ def _print_Allocate(self, expr):
         return '{}\n{}\n{}\n'.format(free_code, shape_Assign, alloc_code)
 
     def _print_Deallocate(self, expr):
-        if isinstance(expr.variable, InhomogeneousTupleVariable):
-            return ''.join(self._print(Deallocate(v)) for v in expr.variable)
-        cuda = ''
-        if expr.variable.on_device or expr.variable.is_managed:
-            cuda = 'cuda_'
-            return ''
+        dealloc_code = ''
         if expr.variable.is_alias:
-            return '{}free_pointer({});\n'.format(cuda, self._print(expr.variable))
-        return '{}free_array({});\n'.format(cuda, self._print(expr.variable))
+            dealloc_code =  f"cuda_free_pointer({expr.variable});"
+        else:
+            if expr.variable.memory_location == 'host':
+                dealloc_code = f"cuda_free_host({expr.variable});"
+            else:
+                dealloc_code = f"cuda_free({expr.variable});"
```

Comment:
You should never print an object using `str(expr)` (this is what you are doing in an f-stream), you need to call the `self._print` function:
```
        var_code = self._print(expr.variable)
        if expr.variable.is_alias:
            dealloc_code =  f"cuda_free_pointer({var_code});"
        elif expr.variable.memory_location == 'host':
            dealloc_code = f"cuda_free_host({var_code});"
        else:
            dealloc_code = f"cuda_free({var_code});"
```

Reply:
Fixed a6341bb0fbb29da2b6290d8421fab40577c423f2